### PR TITLE
 feat: Conn & Stats implemented so that the dbresolver can replace *sql.DB 

### PR DESCRIPTION
**Pull request by goferHiro**
Planning to pass DBImpl into bun

### CODE REVIEW

**Comment by bxcodec** [comment]
Can you help to fix the CI?

File: db.go [hunk 1]
```diff
@@ -33,7 +33,7 @@ type DB interface {
 	SetConnMaxLifetime(d time.Duration)
 	SetMaxIdleConns(n int)
 	SetMaxOpenConns(n int)
-	// Stats() sql.DBStats // not relevant for multi connection DB
+	Stats() sql.DBStats // not relevant for multi connection DB
```
Comment (bxcodec) [inline db.go:36]: please modify the comment, maybe "only for primary db" (?)

**Comment by bxcodec** [comment]
Also, if possible, will you be able to help to test this #11, multi-master connection in your ORM?

File: db.go [hunk 2]
```diff
@@ -17,7 +17,7 @@ type DB interface {
 	Begin() (*sql.Tx, error)
 	BeginTx(ctx context.Context, opts *sql.TxOptions) (*sql.Tx, error)
 	Close() error
-	// Conn(ctx context.Context) (*sql.Conn, error) // not relevant for multi connection DB
+	Conn(ctx context.Context) (*sql.Conn, error) // not relevant for multi connection DB
```
[the remark this answers is not in the document]
Comment (goferHiro) [reply db.go]: okay

File: db.go (hunk 1 above)
Comment (goferHiro) [reply db.go]: Done

**Review by goferHiro — COMMENTED** [review]
I am done

**Comment by goferHiro** [comment]
Hi can we connect via whatsapp or any other means. Chatting here is not so cool

**Comment by goferHiro** [comment]
> Also, if possible, will you be able to help to test this #11, multi-master connection in your ORM?

This is pretty innovative. Definitely I can assist in testing it my orm.  MY PR is currently under review https://github.com/uptrace/bun/pull/729

**Comment by goferHiro** [comment]
> I just realized the struct name is really lame, `DBImpl` fr???? My naming variable capability is questioned now 😓 I want to refactor it like `DatabaseResolver` but it will be a breaking change and need to be released in a major version 😓

I think DatabaseResolver should be the name of a function not a struct. I would suggest using the "multiDBRepo" or "multiDB" or "dbRepo"

Ensuring the struct is not exported is also a wise thing

**Comment by goferHiro** [comment]
> Can you help to fix the CI?

What's your plan?

**Comment by bxcodec** [comment]
> Hi can we connect via whatsapp or any other means. Chatting here is not so cool

Let's use slack. I just created a slack workspace here https://join.slack.com/t/bxcodecoss/shared_invite/zt-1l5yx8iw6-jfwZhU0G7C1F7BAYJRljXQ please join.

Also, please connect with me on LinkedIn [imantumorang](https://www.linkedin.com/in/imantumorang/)

---
> What's your plan?

No, I mean your code is failing on the CI here https://github.com/bxcodec/dbresolver/actions/runs/3646316992/jobs/6157266209. Maybe help to fix it?

---
> Ensuring the struct is not exported is also a wise thing

Looks good, and will reflect it in #11

**Comment by goferHiro** [comment]
Fixed ....

File: go.mod [hunk 3]
```diff
@@ -1,6 +1,6 @@
 module github.com/bxcodec/dbresolver
 
-go 1.17
+go 1.19
```
Comment (bxcodec) [inline go.mod:3]: why does this change exist?
This will be a breaking change. Because it will require at least go 1.19 to be able to use this library

Comment (goferHiro) [reply go.mod:3]: Go 1.19 is backwards compatible

Comment (goferHiro) [reply go.mod:3]: thanks to [Go 1 compatibility promise](https://click.convertkit-mail2.com/n4u9zol3zlfvhmqnvpf6/e0hph7hk9rqmzqb8/aHR0cHM6Ly9nb2xhbmcub3JnL2RvYy9nbzFjb21wYXQ=).

Comment (bxcodec) [reply go.mod:3]: It's backward compatible if I'm using a newer one and the libraries are still using the old one, I can still use the libraries. 
But it won't work if the library is the newer one, but my project uses the older one. I won't be able to use it 🤔 

Need to confirm tho 🤔

Comment (bxcodec) [reply go.mod:3]: I mean, if we really don't need a new feature from go 1.19, it's better to keep it on go 1.17

eg, if we use Generic (go 1.18) then we have to make sure the minimum version in our Go mod is Go 1.18. But we're not using generic here.

**Review by bxcodec — APPROVED** [review]
LGTM

**Pull request merged**